### PR TITLE
Added regularization option to BatchNormalization layer

### PR DESCRIPTION
**Pull request by lolemacs**
Added gamma_regularizer and beta_regularizer parameters to the BatchNormalization layer. Adding a keras.regularizers import was also needed for this.

While not fully mentioned in the original BatchNormalization layer, many recent papers (ResNets and DenseNets being some examples) used regularization on BN layers to achieve their results.

Usage example:
model.add(BatchNormalization(gamma_regularizer=l2(0.001), beta_regularizer=l2(0.001))

### CODE REVIEW

File: keras/layers/normalization.py [hunk 1]
```diff
@@ -57,14 +61,16 @@ class BatchNormalization(Layer):
         - [Batch Normalization: Accelerating Deep Network Training by Reducing Internal Covariate Shift](http://jmlr.org/proceedings/papers/v37/ioffe15.html)
     '''
     def __init__(self, epsilon=1e-5, mode=0, axis=-1, momentum=0.99,
-                 weights=None, beta_init='zero', gamma_init='one', **kwargs):
+                 weights=None, gamma_regularizer=None, beta_regularizer=None, beta_init='zero', gamma_init='one', **kwargs):
```
Comment (fchollet) [inline keras/layers/normalization.py]: New arguments should be added at the very end. Also, need a line break.

**Comment by fchollet** [comment]
Sounds good. This should be added to the BN test.

**Comment by lolemacs** [comment]
Alright, took me a few tries to get pep8 right but it looks right now. I also added regularization tests to the BN test script, like you mentioned.

**Comment by fchollet** [comment]
LGTM
